### PR TITLE
i#1312 AVX-512 support: Add all remaining vperm* opcodes.

### PR DESCRIPTION
**Pull request by hgreving2304**
Adds the evex-promoted opcodes vpermilps, vpermilpd, vpermd, vpermq, vpermps, vpermpd.

Adds the new AVX-512 opcodes vpermw, vpermi2d, vpermi2q, vpermi2ps, vpermi2p, vpermi2b,
vpermi2w, vpermt2b, vpermt2w, vpermt2d, vpermt2q, vpermt2ps, vpermt2pd.

Adds the new size OPSZ_vex32_evex64.

Adds tests for above.

Doubles the encoding buffer size in test api.ir to 16384. Renames disassembly buffer names
"buf" to "dbuf" so it doesn't shadow global encoding buffer "buf".

Opcodes have been checked against llvm-mc, binutils/gas/objdump and capstone.

Issue: #1312

### CODE REVIEW

File: core/arch/decode.h [hunk 1]
```diff
@@ -318,6 +318,7 @@ enum {
     OPSZ_SCALABLE,        /** Scalable size for SVE vector registers. */
     OPSZ_SCALABLE_PRED,   /** Scalable size for SVE predicate registers. */
     OPSZ_16_vex32_evex64, /**< 16, 32, or 64 bytes depending on EVEX.L and EVEX.LL'. */
+    OPSZ_vex32_evex64,    /**< 32, or 64 bytes depending on EVEX.L and EVEX.LL'. */
```
Comment (derekbruening) [inline core/arch/decode.h]: nit: s/,//

File: core/arch/x86/opcode.h [hunk 2]
```diff
@@ -1350,19 +1350,32 @@ enum {
...
     /* 1182 */ OP_vmovdqa32,   /**< IA-32/AMD64 AVX-512 vmovdqa32 opcode. */
     /* 1183 */ OP_vmovdqa64,   /**< IA-32/AMD64 AVX-512 vmovdqa64 opcode. */
-    /* 1184 */ OP_vmovdqu8,    /**< IA-32/AMD64 AVX-512 vmovdqu8 opcode. */
-    /* 1185 */ OP_vmovdqu16,   /**< IA-32/AMD64 AVX-512 vmovdqu16 opcode. */
-    /* 1186 */ OP_vmovdqu32,   /**< IA-32/AMD64 AVX-512 vmovdqu32 opcode. */
-    /* 1187 */ OP_vmovdqu64,   /**< IA-32/AMD64 AVX-512 vmovdqu64 opcode. */
+    /* 1184 */ OP_vmovdqu16,   /**< IA-32/AMD64 AVX-512 vmovdqu16 opcode. */
+    /* 1185 */ OP_vmovdqu32,   /**< IA-32/AMD64 AVX-512 vmovdqu32 opcode. */
+    /* 1186 */ OP_vmovdqu64,   /**< IA-32/AMD64 AVX-512 vmovdqu64 opcode. */
+    /* 1187 */ OP_vmovdqu8,    /**< IA-32/AMD64 AVX-512 vmovdqu8 opcode. */
```
Comment (derekbruening) [inline core/arch/x86/opcode.h]: This re-ordering seems inferior: can we sort with -n?

File: core/arch/decode.h (hunk 1 above)
Comment (hgreving2304) [reply core/arch/decode.h]: Fixed.

File: core/arch/x86/opcode.h (hunk 2 above)
Comment (hgreving2304) [reply core/arch/x86/opcode.h]: Done.

**Pull request merged**